src/Foundation/Commerce/code/Services/Billing/BillingService.cs

```diff
@@ -15,28 +15,233 @@
 namespace Wooli.Foundation.Commerce.Services.Billing
 {
     using System;
+    using System.Collections.Generic;
+    using System.Linq;
+
+    using Connect.Context;
+    using Connect.Managers.Payment;
+
+    using Context;
 
     using Base.Models;
 
+    using Connect.Managers.Cart;
+    using Connect.Models;
+
     using DependencyInjection;
 
-    using Models;
-    using Models.Entities;
+    using Mappers.Payment;
+
     using Models.Entities.Addresses;
     using Models.Entities.Billing;
-    using Models.Entities.Payment;
+
+    using Sitecore.Commerce.Entities.Carts;
+    using Sitecore.Diagnostics;
+
+    using Sitecore.Commerce.Entities.Payments;
+    using Sitecore.Commerce.Services.Carts;
+
+    using FederatedPaymentInfo = Models.Entities.Payment.FederatedPaymentInfo;
+    using PaymentOption = Models.Entities.Payment.PaymentOption;
+    using PaymentMethod = Models.Entities.Payment.PaymentMethod;
 
     [Service(typeof(IBillingService), Lifetime = Lifetime.Singleton)]
     public class BillingService : IBillingService
     {
-        public Result<BillingInfo> GetBillingOptions()
+        private readonly ICartManagerV2 cartManager;
+        private readonly IPaymentMapper paymentMapper;
+        private readonly IPaymentManagerV2 paymentManager;
+        private readonly IStorefrontContext storefrontContext;
+        private readonly IVisitorContext visitorContext;
+
+        public BillingService(
+            ICartManagerV2 cartManager,
+            IStorefrontContext storefrontContext,
+            IVisitorContext visitorContext,
+            IPaymentMapper paymentMapper,
+            IPaymentManagerV2 paymentManager)
+        {
+            Assert.ArgumentNotNull(cartManager, nameof(cartManager));
+            Assert.ArgumentNotNull(paymentMapper, nameof(paymentMapper));
+            Assert.ArgumentNotNull(paymentManager, nameof(paymentManager));
+            Assert.ArgumentNotNull(storefrontContext, nameof(storefrontContext));
+            Assert.ArgumentNotNull(visitorContext, nameof(visitorContext));
+
+            this.cartManager = cartManager;
+            this.paymentMapper = paymentMapper;
+            this.storefrontContext = storefrontContext;
+            this.visitorContext = visitorContext;
+            this.paymentManager = paymentManager;
+        }
+
+        public Result<BillingInfo> GetBillingInfo()
+        {
+            var model = new BillingInfo();
+            var result = new Result<BillingInfo>(model);
+
+            if (!Sitecore.Context.PageMode.IsExperienceEditor)
```

Comment:
How to cover this one with unit tests?

Reply:
We could try something like https://jockstothecore.com/how-to-test-automation/